nodes/src/nodes/tool_exa_search/exa_driver.py

```
@@ -0,0 +1,372 @@
+
+        query = input_obj.get('query')
+        if not query or not isinstance(query, str) or not query.strip():
+            raise ValueError('query is required and must be a non-empty string')
+
+        search_type = input_obj.get('type')
+        if search_type is not None and search_type not in VALID_SEARCH_TYPES:
+            raise ValueError(f'type must be one of {sorted(VALID_SEARCH_TYPES)}; got {search_type!r}')
+
+        num_results = input_obj.get('num_results')
+        if num_results is not None:
+            if not isinstance(num_results, int) or num_results < 1 or num_results > 50:
```

Ruff (0.15.9)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)